src/mem.rs

```diff
@@ -58,3 +62,105 @@ pub unsafe extern "C" fn memcmp(s1: *const u8, s2: *const u8, n: usize) -> i32 {
     }
     0
 }
+
+fn memcpy_element_unordered_atomic<T: Copy>(dest: *mut T, src: *const T, bytes: usize) {
+    unsafe {
+        let n = unchecked_div(bytes, mem::size_of::<T>());
```

Comment:
Can we assume that `bytes` is a multiple of `size_of::<T>`? If not, what happens when there is rounding going on?

Reply:
Yes, this guaranteed by LLVM:
> [If len is not a positive integer multiple of element_size, then the behaviour of the intrinsic is undefined.](https://llvm.org/docs/LangRef.html#id942)

Reply:
Seems like you could use https://doc.rust-lang.org/nightly/std/intrinsics/fn.exact_div.html then. Also IMO this is a precondition worth stating in a comment here.

Reply:
Ooh, didn't see that one. Thanks!

Reply:
(Resolved in https://github.com/rust-lang/compiler-builtins/pull/312.)